main.c

```diff
@@ -80,9 +82,9 @@ void uppercase(char* str) {
 int compare_alphabetic(const void* v1, const void* v2) {
     const struct window_props* p1 = v1;
     const struct window_props* p2 = v2;
-    lowercase(p1->name);
-    lowercase(p2->name);
-    return strcmp(p1->name, p2->name);
+    lowercase(p1->class);
```

Comment:
This was something I couldn't think of beforehand - So are you sure you want to sort by *class* alphabetically, when using the title option?

Actually I do see how this avoids one problem: when a title changes, it would re-sort the list which could make you lose track of which window was which... Yeah, maybe this is indeed good?

Do you use the alphabetic sorting option btw?

Reply:
Ah, so you did run into what I was talking about and it was awful to use? That's so good to know.

Well, if we want the alphabetic sorting to literally mean sort alphabetically of whatever string the window name is, that's just how it's gonna be.

Maybe the sorting by class in this case is definitely more usable - at least what it achieves is a loose grouping by application. But when the class is not visible, it's a bit vague to the user what `alphabetic` means.

Hmm, gives me the idea that `sort="alphabetic"` could be reworded to `sort="by_application"` or something along those lines.

If you have an idea/feeling of what the perfect sorting would look like for you, please describe.

Reply:
Yeah this seems very clear, I suggest lets rename it:

config `sort_by="alphabetic"` -> `sort_by="application"``

Then lets follow-up after PR if there are good ideas for sorting with `title` on, okay?

Reply:
Also you may already know, but maybe it helps clarify this too:

The unsorted `sort_by="none"` in most window managers means oldest window first, newest last. I'm not sure if there are exceptions to that, though I vaguely remember reading something like that online. That technically a WM can get the client list in some other order..... but is there any WM that actually does? Probably not....?

Reply:
> I suggest lets rename it
> sort_by="application"

Nice and clear.
I feel for now this is the best we can do until we figure out how to properly listen to `WM_NAME` update events, which we might move to a new discussion if you agree.

Reply:
Yeap! Thanks a lot, I'm ready to merge your PR.

I wanna track more clearly what is being worked on so I made this issue: https://github.com/tuurep/windowlist/issues/6

And I have another open for my "small id for duplicate windows" idea.

You can join in on the issues or use the discussions, or however is most convenient :smile: